packages/core/test/guards.test.ts

```diff
@@ -107,9 +107,11 @@ describe('guard conditions', () => {
 
   it('should not transition if no condition is met', () => {
     const nextState = lightMachine.transition(
-      State.from('green', {
-        elapsed: 9000
-      }),
+      lightMachine.resolveState(
```

Comment:
q: in this file you wrap all `State.from` with `machine.resolveState` - this is correct and better, i just wonder if there is any particular logic in this PR that has made this obligatory right now?

Reply:
Yes, `state.done` needs to read the machine from `state.configuration[0].machine` but `state.configuration` is empty from `State.from(...)`; `machine.resolve` provides that.

Same if we want to read from `state.machine` - that does not exist in `State.from(...)`

Reply:
This PR checks `state.done` in the `while` loop, and it will throw an error without this.